### PR TITLE
Implement CLI `--total` functionality

### PR DESCRIPTION
**Pull request by mhmohona**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

Implemented the `--total (-t)` functionality would check Wikidata for the total of certain groupings of languages and word types.

image

### Related issue

Fixes - #147

### CODE REVIEW

**Comment by mhmohona** [comment]
It still has a problem. It is recognizing data-type as language like this - 
image

**Comment by andrewtavis** [comment]
Any thoughts on what's causing it, @mhmohona? :)

File: src/scribe_data/cli/total.py [hunk 1]
```diff
@@ -19,3 +19,93 @@
...
+    "german": "Q188",
+    "italian": "Q652",
+    "portuguese": "Q5146",
+    "russian": "Q7737",
+    "spanish": "Q1321",
+    "swedish": "Q9027"
+}
+
+# Dictionary to map data types to their Wikidata Q-IDs
+data_type_to_qid = {
```
Comment (mhmohona) [inline src/scribe_data/cli/total.py:38]: @andrewtavis will you please check here if I have the correct QID?

File: src/scribe_data/cli/total.py [hunk 2]
```diff
@@ -19,3 +19,93 @@
...
+        The input string representing a language or data type.
+
+    Returns
+    -------
+    str or None
+        The QID corresponding to the input string, or None if not found.
+    """
+    if input_str:
+        input_str_lower = input_str.lower()
+        if input_str_lower in language_to_qid:
```
Comment (andrewtavis) [inline src/scribe_data/cli/total.py:61]: Putting a space before all `else` statements and maybe between the `elif` and the `return` would be good :)

**Comment by andrewtavis** [comment]
Minor comments so far:
- Let's include the language and the data type in the output
- From @wkyoshida, something like:

```bash
Language: German
Data type: Verbs
Total number of lexemes: 999
```

- Let's move `data_type_to_qid` to the `data_type_metadata.json` file
- Let's reference the `language_metadata.json` file for the `language_to_qid` information

We'll be good to go after all this! 🥳

**Comment by andrewtavis** [comment]
Quick check here @mhmohona, are you planning on getting to the changes we mentioned above?

**Comment by wkyoshida** [comment]
> It still has a problem. It is recognizing data-type as language like this - image

Doesn't this work since `-l German` is passed first? Could just be that the first argument is recognized and used. The second may have been passed, but it's disregarded. 

Looks like there is a [choices](https://docs.python.org/3/library/argparse.html#choices) parameter that we can pass in to `add_argument()` to specify what are the allowable options for an argument. Maybe we look if specifying this could make sense as the mechanism for controlling valid/invalid inputs?

**Comment by mhmohona** [comment]
Currently stuck in this comment 😢 
> Let's move data_type_to_qid to the data_type_metadata.json file

**Comment by andrewtavis** [comment]
By this I mean let's move the functionality of `data_type_to_qid` to the `data_type_metadata.json` file such that we just import the data at the top of the file rather than using a custom function :)

**Comment by mhmohona** [comment]
> > It still has a problem. It is recognizing data-type as language like this - image
> 
> Doesn't this work since `-l German` is passed first? Could just be that the first argument is recognized and used. The second may have been passed, but it's disregarded.
> 
> Looks like there is a [choices](https://docs.python.org/3/library/argparse.html#choices) parameter that we can pass in to `add_argument()` to specify what are the allowable options for an argument. Maybe we look if specifying this could make sense as the mechanism for controlling valid/invalid inputs?

@wkyoshida, thank you for looking into it. I have solved this problem.

**Comment by mhmohona** [comment]
> By this I mean let's move the functionality of `data_type_to_qid` to the `data_type_metadata.json` file such that we just import the data at the top of the file rather than using a custom function :)

I need help with the QID. :( Unable  to find the correct QIDs. It would be super helpful for me if you would update the `data_type_metadata.json` file with QIDs @andrewtavis.

File: src/scribe_data/cli/main.py [hunk 3]
```diff
@@ -127,7 +128,11 @@ def main() -> None:
     )
     total_parser._actions[0].help = "Show this help message and exit."
     total_parser.add_argument(
-        "-lang", "--language", type=str, help="The language(s) to check totals for."
+        "-l",
```
Comment (andrewtavis) [inline src/scribe_data/cli/main.py:131]: I'm going to remove `-l` as an option, @mhmohona, as I think that making sure that we don't have commands and arguments that are the same is important :)

File: src/scribe_data/cli/main.py [hunk 4]
```diff
@@ -190,7 +195,13 @@ def main() -> None:
         )
 
     elif args.command in ["total", "t"]:
-        return
+        if not args.language and not args.data_type:
```
Comment (andrewtavis) [inline src/scribe_data/cli/main.py:198]: Really appreciate the quality of your messages to the user, @mhmohona!

File: src/scribe_data/cli/main.py [hunk 5]
```diff
@@ -190,7 +195,13 @@ def main() -> None:
...
+            print(
+                "Error: At least one of -lang/--language or -dt/--data-type must be specified."
+            )
+            total_parser.print_help()
+            return
```
Comment (andrewtavis) [inline src/scribe_data/cli/main.py:203]: Let's always do a space after `return` :)

**Review by andrewtavis — APPROVED** [review]
Thanks so much for this, @mhmohona! The full experience of Scribe-Data is really coming along 😊

**Pull request merged**